src/api/app.js

```diff
@@ -46,3 +65,45 @@ app.get('/', function (req, res) {
 
 const port = 3000;
 app.listen(port, () => logger.info(`Start service on port ${port}!`));
+
+function removeEmailIfOlderThan(email, dateTimeBoundary) {
+    return email["datetime"] > dateTimeBoundary;
+}
+
+function filterByEmail(email, to) {
+    let actualToEmail = email["personalizations"][0]["to"][0]["email"];
+    return actualToEmail === to;
+}
+
+function filterBySubject(email, subject) {
+    let actualSubject = email["subject"];
+    if(subject.startsWith('%') && subject.endsWith('%')) {
+        searchSubject = subject.substring(1, subject.length - 1);
+        console.log('Searching for emails with subject containing', searchSubject);
+        return actualSubject.includes(searchSubject);
+    }
+    return actualSubject === subject;
+}
+
+function filterByDateTimeSince(email, dateTimeSinceAsString) {
+    let dateTimeSince = Date.parse(dateTimeSinceAsString);
+    if(dateTimeSince === NaN) {
```

Comment:
Please consider <https://eslint.org/docs/rules/use-isnan>:

```suggestion
    if(isNaN(dateTimeSince)) {
```